app/views/NewServerView.js

```diff
@@ -128,21 +134,14 @@ export default class NewServerView extends LoggedView {
 	}
 
 	completeUrl = (url) => {
+		const { protocolType } = this.state;
 		url = url && url.trim();
 
 		if (/^(\w|[0-9-_]){3,}$/.test(url)
-			&& /^(htt(ps?)?)|(loca((l)?|(lh)?|(lho)?|(lhos)?|(lhost:?\d*)?)$)/.test(url) === false) {
+			&& /^(loca((l)?|(lh)?|(lho)?|(lhos)?|(lhost:?\d*)?)$)/.test(url) === false) {
 			url = `${ url }.rocket.chat`;
 		}
-
-		if (/^(https?:\/\/)?(((\w|[0-9])+(\.(\w|[0-9-_])+)+)|localhost)(:\d+)?$/.test(url)) {
-			if (/^localhost(:\d+)?/.test(url)) {
-				url = `http://${ url }`;
-			} else if (/^https?:\/\//.test(url) === false) {
-				url = `https://${ url }`;
-			}
-		}
-
+		url = protocolType + url;
```

Comment:
prefer using templates instead of concatenation

Reply:
cool

Reply:
maybe += is better

Reply:
not critical at all, I just prefer using templates when working with strings (and templates are used all around in this project)